source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+                    // interpolate between y = height and upper airfoil points
+                    // (block2) or y = -height and lower airfoil points (block5)
+                    node = airfoil_1D[(
+                             (cell.material_id() == id_block_2) ? (0) : (1))]
+                                     [n_cells_x_0 + ix] *
+                             alpha +
+                           p * (1 - alpha);
+                  }
+                else if (cell.material_id() == id_block_3 ||
+                         cell.material_id() == id_block_6) // block 3 and 6
+                  {
+                    // compute indices ix and iy
+                    const double dx = length_b2 / n_cells_x_2;
+                    const double dy = height / n_cells_y;
+                    const int    ix = static_cast<int>(
+                      std::round(std::abs(node(0) - H(0)) / dx));
+                    const int iy =
+                      static_cast<int>(std::round(std::abs(node(1)) / dy));
+
+                    const double alpha_y = bias_alpha(1 - 1.0 * iy / n_cells_y);
+                    const double alpha_x =
+                      bias_alpha(1 - (static_cast<double>(ix)) / n_cells_x_2);
+                    // define on upper/lower horizontal far field side at y =
+                    // +/- height, i.e. face GK or IL incline factor to move
+                    // points G and H to the right
+                    const Point<2> p1(J(0) - (1 - incline_factor) * length_b2 *
+                                               (alpha_x),
+                                      ((cell.material_id() == id_block_3) ?
+                                         (height) :
+                                         (-height)));
+                    // define points on HJ but use tail_y as y-coordinate, in
+                    // case last airfoil point has y =/= 0
+                    const Point<2> p2(J(0) - alpha_x * length_b2, tail_y);
+                    node = p1 * (1 - alpha_y) + p2 * alpha_y;
+                  }
+                else
+                  {
+                    Assert(false,
+                           ExcIndexRange(cell.material_id(),
+                                         id_block_1,
+                                         id_block_6));
+                  }
+              }
+        }
+
+
+        /*
+         * This function returns a bias factor 'alpha' which is used to make the
+         * mesh more tight in close distance of the airfoil.
+         * It is a bijective function mapping from [0,1] onto [0,1] where values
+         * near 1 are made tighter.
+         */
+        double
+        bias_alpha(double alpha) const
+        {
+          return std::tanh(bias_factor * alpha) / std::tanh(bias_factor);
+        }
+      };
+    } // namespace
+
+
+    /**
+     * Initialize the given triangulation an airfoil, i.e. a Mesh of C-Type
+     * approximating Joukowski or NACA (4 digit) airfoils.
```

Comment:
Is this text just a copy of the documentation you have in the .h file? If so, just delete it here -- it's in the .h file where doxygen can pick it up, and having duplication is always a prescription for letting things get out of sync.

Reply:
It's not a copy but it states exactly the same.  There is only this additional info in that documentation:
 `By default, a triangulation for a NACA 2412 airfoil with chord length one is generated.` 
Is this information necessary?